tests/subsys/net/lib/nrf_cloud/nrf_cloud_test/src/nrf_cloud_test.c

```diff
@@ -0,0 +1,609 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <unity.h>
+#include <stdio.h>
+#include <net/nrf_cloud.h>
+#include <nrf_cloud_transport.h>
+#include <nrf_cloud_fsm.h>
+#include <nrf_cloud_codec.h>
+#include <zephyr/fff.h>
+
+DEFINE_FFF_GLOBALS;
+
+/* Fake functions declaration */
+FAKE_VALUE_FUNC(int, nct_init, const char *);
+FAKE_VALUE_FUNC(int, nfsm_init);
+FAKE_VALUE_FUNC(int, nrf_cloud_codec_init);
+FAKE_VALUE_FUNC(int, nrf_cloud_fota_fmfu_dev_set, const struct dfu_target_fmfu_fdev *);
+FAKE_VALUE_FUNC(int, nct_disconnect);
+FAKE_VOID_FUNC(nct_uninit);
+FAKE_VALUE_FUNC(int, nrf_cloud_fota_uninit);
+FAKE_VALUE_FUNC(int, nct_connect);
+FAKE_VALUE_FUNC(int, nct_socket_get);
+FAKE_VALUE_FUNC(int, nct_keepalive_time_left);
+FAKE_VALUE_FUNC(int, nct_process);
+
+/* This is required since unity_main return int and
+ * zephyr expects main not to return any value
+ */
+extern int unity_main(void);
+
+/* Simple event handler needed for the init function to
+ * have a success initialization in the test below
+ */
+void event_handler(const struct nrf_cloud_evt *const evt)
+{
+	switch (evt->type) {
+	case NRF_CLOUD_EVT_READY:
+		printk("Device is connected and ready");
+		break;
+	case NRF_CLOUD_EVT_ERROR:
+		printk("Failed to connect to cloud");
+		break;
+	default:
+		break;
+	}
+}
+
+int nct_init_success(const char *const client_id)
+{
+	return 0;
+}
+
+int nct_init_fail(const char *const cliend_id)
+{
+	return -ENODEV;
+}
+
+int nfsm_init_success(void)
+{
+	return 0;
+}
+
+int nrf_cloud_codec_init_success(void)
```

Comment:
I think `succeeds` instead of `success` would improve readability

I also recommend you use a double underscore before `success` or `succeeds` to distinguish it from the mocked-name `nsfm_cloud_codec_init`